### PR TITLE
test: change all tests in the circuit test path to use with pytest.raises

### PR DESCRIPTION
**Pull request by math411**
*Issue #, if available:*

*Description of changes:*
fix: change all tests in the circuit test path to use with pytest.raises

*Testing done:*
- Blindly believed that it would work
- Trusted the process
- Ran `tox -e unit-tests`

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by speller26 — APPROVED** [review]

File: src/braket/circuits/braket_program_context.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ def add_result(self, result: Results) -> None:
         """
         self._circuit.add_result_type(braket_result_to_result_type(result))
 
-    def handle_parameter_value(self, value):
+    def handle_parameter_value(self, value: Any) -> Any:
```
Comment (ajberdy) [inline src/braket/circuits/braket_program_context.py:137]: This should actually be `Union[float, Expr] -> Any` in the base class and `Union[float, Expr] -> Union[float, FreeParameterExpression]` here.

I'll update the base class in the local sim

Comment (ajberdy) [reply src/braket/circuits/braket_program_context.py:137]: (don't block merging this PR on this, I can update it separately)

**Review by ajberdy — APPROVED** [review]

File: src/braket/circuits/braket_program_context.py (hunk 1 above)
Comment (math411) [reply src/braket/circuits/braket_program_context.py:137]: @ajberdy I think this will also fail linters for the description

Comment (ajberdy) [reply src/braket/circuits/braket_program_context.py:137]: It gets its docstring from the base class in the default simulator. Wouldn't hurt to give a more specific one here though

**Pull request merged**